src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CompilationContext.cs

```diff
@@ -1487,6 +1499,22 @@ private static bool IsTransparentIdentifierFieldInAnonymousType(FieldSymbol fiel
             }
         }
 
+        private static bool TryParseGeneratedName(string name, out GeneratedNameKind kind, out string part)
```

Comment:
Will leave name as is so it's clear the method does not assume the name can be parsed.